corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -107,8 +107,9 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
     var showMenu = function (menuResponse) {
         var menuListView = menusUtils.getMenuView(menuResponse);
         var appPreview = FormplayerFrontend.currentUser.displayOptions.singleAppMode;
-        var sidebarEnabled = toggles.toggleEnabled('SPLIT_SCREEN_CASE_SEARCH') && !appPreview;
-
+        var queryResponse = menuResponse.queryResponse;
+        const hasSearchInputs = menuResponse.type === "query" ? menuResponse.length > 0 : (menuResponse.queryResponse != null && menuResponse.queryResponse.displays.length > 0);
```

Comment:
nit: this is ugly but I don't have a better idea-- I've been wanting to refactor these checks for the next like 20 lines for a while so maybe this is just my personal reminder

Reply:
+1

Reply:
I agree. I don't have any great ideas at the moment, but it is a bit less cluttered after https://github.com/dimagi/commcare-hq/pull/33876/commits/47a1c1342e5e12dc1796f29c3625e2728607b6a7